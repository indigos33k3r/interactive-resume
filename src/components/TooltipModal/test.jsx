import React from 'react'
import {shallow} from 'enzyme'

describe('TooltipModal', () => {
  it('renders <TooltipModal />', () => {
    const component = render()

    expect(component).toMatchSnapshot()
  })

  context('when <button /> was clicked', () => {
    let component

    beforeEach(() => {
      component = render()
      component.find('button').prop('onClick')()
      component.update()
    })

    it('shows modal', () => {
      expect(component).toMatchSnapshot()
    })

    context('when <Modal /> #prop.onExited was called', () => {
      it('hides modal', () => {
        component.find('Modal').prop('onExited')()
        component.update()

        expect(component).toMatchSnapshot()
      })
    })
  })

  function render(props = {}) {
    const defaultProps = {
      tooltipText: 'Some tooltip text',
      children: 'Some children text',
      modalClass: 'Some modal class',
      additionalClass: 'Some tooltip class',
      enterCallback: () => {}
    }
    const TooltipModal = require('.').default

    return shallow(<TooltipModal {...defaultProps} {...props} />)
  }
})
