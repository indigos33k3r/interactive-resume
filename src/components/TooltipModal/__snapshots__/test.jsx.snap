// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipModal renders <TooltipModal /> 1`] = `
<Fragment>
  <button
    className="tooltip Some tooltip class"
    onClick={[Function]}
  >
    <div
      className="tooltip__exclamation"
    >
      !
    </div>
    <div
      className="tooltip__text-box"
    >
      <div
        className="tooltip__text"
      >
        Some tooltip text
      </div>
    </div>
    <div
      className="tooltip__arrow"
    />
  </button>
</Fragment>
`;

exports[`TooltipModal when <button /> was clicked shows modal 1`] = `
<Fragment>
  <button
    className="tooltip Some tooltip class"
    onClick={[Function]}
  >
    <div
      className="tooltip__exclamation"
    >
      !
    </div>
    <div
      className="tooltip__text-box"
    >
      <div
        className="tooltip__text"
      >
        Some tooltip text
      </div>
    </div>
    <div
      className="tooltip__arrow"
    />
  </button>
  <Modal
    modalClass="Some modal class"
    onEnter={[Function]}
    onExited={[Function]}
  >
    Some children text
  </Modal>
</Fragment>
`;

exports[`TooltipModal when <button /> was clicked when <Modal /> #prop.onExited was called hides modal 1`] = `
<Fragment>
  <button
    className="tooltip Some tooltip class"
    onClick={[Function]}
  >
    <div
      className="tooltip__exclamation"
    >
      !
    </div>
    <div
      className="tooltip__text-box"
    >
      <div
        className="tooltip__text"
      >
        Some tooltip text
      </div>
    </div>
    <div
      className="tooltip__arrow"
    />
  </button>
</Fragment>
`;
